$OpenBSD$

GHC 9.10+ compat: add SA.Unzip KeyMap instance (required by
semialign>=1.3), add fst,snd to shared Prelude import (available
to both CPP branches), relax ghc-prim upper bound.

--- aeson.cabal
+++ aeson.cabal
@@ -100,7 +100,7 @@
     , containers        >=0.5.10.2 && <0.7
     , deepseq           >=1.4.3.0  && <1.6
     , exceptions        >=0.10.4   && <0.11
-    , ghc-prim          >=0.5.0.0  && <0.12
+    , ghc-prim          >=0.5.0.0  && <0.14
     , template-haskell  >=2.12.0.0 && <2.22
     , text              >=1.2.3.0  && <1.3  || >=2.0 && <2.2
     , time              >=1.8.0.2  && <1.13
--- src/Data/Aeson/KeyMap.hs
+++ src/Data/Aeson/KeyMap.hs
@@ -100,6 +100,7 @@
 import Prelude ((.), ($))
 import Prelude (Functor(fmap), Monad(..))
 import Prelude (Show, showsPrec, showParen, shows, showString)
+import Prelude (fst, snd)
 
 import Control.Applicative (Applicative)
 import Control.DeepSeq (NFData(..))
@@ -351,7 +352,6 @@
 
 import Data.List (sortBy)
 import Data.Ord (comparing)
-import Prelude (fst)
 
 -- | A map from JSON key type 'Key' to 'v'.
 newtype KeyMap v = KeyMap { unKeyMap :: HashMap Key v }
@@ -693,6 +693,9 @@
 instance SA.Zip KeyMap where
     zipWith = intersectionWith
 
+
+instance SA.Unzip KeyMap where
+    unzip (KeyMap m) = (KeyMap (fmap fst m), KeyMap (fmap snd m))
 instance SAI.ZipWithIndex Key KeyMap where
     izipWith = intersectionWithKey
 
